packages/common/src/activity-options.ts

```diff
@@ -86,3 +86,49 @@ export interface ActivityFunction<P extends any[], R> {
  * Mapping of Activity name to function
  */
 export type ActivityInterface = Record<string, ActivityFunction<any[], any>>;
+
+/**
+ * Transforms type `T` into an {@link ActivityInterface} by extracting all activity functions.
+ */
+// eslint-disable-next-line @typescript-eslint/ban-types
+export type IntoActivityInterface<T extends object> = {
+  [P in keyof T]: T[P] extends ActivityFunction<any[], any> ? T[P] : never;
+};
+
+/**
+ * Get all defined methods of an object prototype.
+ *
+ * (Taken from https://github.com/sindresorhus/auto-bind/blob/bfdc7861c61d78969912cb2a2d620cdc18de079c/index.js)
+ */
+// eslint-disable-next-line @typescript-eslint/ban-types
+function getPrototypeMethods(prototype: object) {
+  const properties = new Set<string | symbol>();
+
+  for (const key of Reflect.ownKeys(prototype)) {
+    if (key === 'constructor') continue;
```

Comment:
Do we also need to ignore `__proto__` here? For functions, `typeof fn.__proto__ === 'function'`.

Reply:
Not AFAICT according to the source, but we might need to do the traversal that's done here:
https://github.com/sindresorhus/auto-bind/blob/bfdc7861c61d78969912cb2a2d620cdc18de079c/index.js#L11